compiler-core/src/language_server/tests/compilation.rs

```diff
@@ -1,95 +1,157 @@
-
-    drop(engine);
-    let actions = io.into_actions();
     assert_eq!(
-        actions,
-        vec![
-            // new
-            Action::DependencyDownloadingStarted,
-            Action::DownloadDependencies,
-            Action::DependencyDownloadingFinished,
-            Action::LockBuild,
-            Action::UnlockBuild,
-            // compile_please
-            Action::CompilationStarted,
-            Action::LockBuild,
-            Action::UnlockBuild,
-            Action::CompilationFinished,
-        ]
-    )
+        response.compiled_modules,
+        vec![std::path::PathBuf::from("/src/app.gleam")]
+    );
+}
+
+#[test]
+fn compile_project_with_warnings() {
+    let io = LanguageServerTestIO::new();
+    let mut engine = setup_language_server(&io);
+
+    let program_with_warnings = r#"
+        pub fn main() {}
+    "#;
+    io.write_source_module("app", program_with_warnings);
+
+    let response = engine.compile_please();
+
+    assert!(response.result.is_ok());
+    assert_eq!(response.warnings.len(), 1);
+    assert_eq!(response.compiled_modules.len(), 1);
+}
+
+#[test]
+fn compile_multiple_files() {
+    let io = LanguageServerTestIO::new();
+    let mut engine = setup_language_server(&io);
+
+    let app_file = r#"
+        import app2
+        pub fn main() { app2.main() }
+    "#;
+    let app_2_file = r#"
+        pub fn main() { True }
+    "#;
+    io.write_source_module("app", app_file);
+    io.write_source_module("app2", app_2_file);
+
+    let response = engine.compile_please();
+
+    assert!(response.result.is_ok());
+    assert_eq!(response.compiled_modules.len(), 2);
+}
+
+#[test]
+fn compile_test_file() {
+    let io = LanguageServerTestIO::new();
+    let mut engine = setup_language_server(&io);
+
+    let app_file = r#"
+        import app2
+        pub fn main() { app2.main() }
+    "#;
+    let app_2_file = r#"
+        pub fn main() { True }
+    "#;
+    let app_test_file = r#"
+        import app
+        pub fn main() { app.main() == True }
+    "#;
+
+    io.write_source_module("app", app_file);
+    io.write_source_module("app2", app_2_file);
+    io.write_test_module("app_test", app_test_file);
+
+    let response = engine.compile_please();
+
+    assert!(response.result.is_ok());
+    assert_eq!(response.compiled_modules.len(), 3);
+}
+
+#[test]
+fn compile_empty_project() {
+    let io = LanguageServerTestIO::new();
+    let mut engine = setup_language_server(&io);
+
+    let response = engine.compile_please();
+
+    assert!(response.result.is_ok());
+    assert!(response.warnings.is_empty());
+    assert!(response.compiled_modules.is_empty());
 }
 
 #[test]
 fn compile_error_in_src() {
     let io = LanguageServerTestIO::new();
-    let mut engine = setup_engine(&io);
+    let mut engine = setup_language_server(&io);
 
-    io.src_module("app/error", "pub type Error {");
+    io.write_source_module("app/error", "pub type Error {");
 
     let response = engine.compile_please();
+
```

Comment:
Revert the deletion of the remaining assertions please.

Reply:
Were you wanting each test method to have the action assertions, or just the error cases here?

Reply:
Up to you. :)